### PR TITLE
Proj0029: Use C# specific properties only when applicable

### PR DESCRIPTION
**Pull request by Corniel**
Equivalent to #223, but for C#. Currently, only 3 properties are involved.

### CODE REVIEW

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/UseInCSharpContextOnly.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        if (!InCSharpContext(context.File))
+        {
+            Walk(context.File, context);
+        }
+    }
+
+    private static bool InCSharpContext(MsBuildProject project) => project.FileType switch
+    {
+        ProjectFileType.ProjectFile => project.Path.Extension.IsMatch(".csproj"),
+        ProjectFileType.DirectoryBuild => !project.Path.Directory.Files("*.vbproj").Any(),
```
Comment (LauraXiulan) [inline src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/UseInCSharpContextOnly.cs:26]: In the vbonly rule you check if there are any vbproj files, shouldn't you check here if there are any csproj files

File: src/DotNetProjectFile.Analyzers/Rule.cs [hunk 2]
```diff
@@ -254,6 +254,16 @@ public static class Rule
         tags: ["conditions"],
         category: Category.Reliability);
 
+    public static DiagnosticDescriptor UseInCSharpContextOnly => New(
+        id: 0029,
+        title: "Use C# specific properties only when applicable",
+        message: "The property <{0}> is only applicable when using C# and can therefor be removed.",
+        description:
+            "Properties only applicable to VB.NET are noise when none of " +
+            "the involved targets is a VB.NET target.",
```
Comment (LauraXiulan) [inline src/DotNetProjectFile.Analyzers/Rule.cs:263]:
```suggestion
            "Properties only applicable to C# are noise when none of " +
            "the involved targets is a C# target.",
```

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/UseInCSharpContextOnly.cs (hunk 1 above)
Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/UseInCSharpContextOnly.cs]: No. There are more extension types which can be other languages, but also other C# based types, where there is (as far as I know) only one type of project file extension linked to VB.NET. So hence the asymmetrical implementation.

File: src/DotNetProjectFile.Analyzers/DotNetProjectFile.Analyzers.csproj [hunk 3]
```diff
@@ -28,6 +28,7 @@
     <PackageReleaseNotes>
       <![CDATA[
 ToBeReleased
+- Proj0029: Use C# specific properties only when applicable. (NEW RULE)
```
Comment (Gemberkoekje) [inline src/DotNetProjectFile.Analyzers/DotNetProjectFile.Analyzers.csproj:31]: Should the version be updated?

**Review by Gemberkoekje — APPROVED** [review]

File: src/DotNetProjectFile.Analyzers/DotNetProjectFile.Analyzers.csproj (hunk 3 above)
Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/DotNetProjectFile.Analyzers.csproj:31]: Not yet.

**Pull request merged**